paperqa/llms.py

```diff
@@ -802,8 +806,25 @@ async def similarity_search(
     def clear(self) -> None:
         self.texts_hashes = set()
 
+    async def partitioned_similarity_search(
+        self,
+        query: str,
+        k: int,
+        embedding_model: EmbeddingModel,
+        partitioning_fn: Callable[[Doc], int],
+    ) -> tuple[Sequence[Embeddable], list[float]]:
+        raise NotImplementedError(
+            "Need to use the partitioning_fn argument to "
+            "stratify the similarity search."
```

Comment:
I feel like it's weird to use this wording in the base class, it can be misleading. It seems like a combination of:
- An`Args:` description for `partitioning_fn`
- An `abstractmethod`'s `NotImplementedError`

Can we move to:
1. A docstring with `Args:` (especially since this is an `abstractmethod`)
2. A more intuitive message in the `NotImplementedError`

Reply:
good call -- added a full docstring here.